### ISSUE TITLE
ACCOUNT_ID Override is not working

### ISSUE THREAD

**Issue by bsneider** [spulec/moto#5027]
Moto version's tried master branch and 3.1.4

```python
@mock_sts
def test_sts_account_override(monkeypatch):
    dummy_account_id = "MEMEMEMEME"
    monkeypatch.setenv("MOTO_ACCOUNT_ID", dummy_account_id)
    os_account_id = os.environ.get("MOTO_ACCOUNT_ID")
    os_account_id.should.equal(dummy_account_id)
    client = boto3.client("sts", region_name="us-west-2")
    resp = client.get_caller_identity()
    resp["Account"].should.equal(dummy_account_id)
```
this fails. 

Expected behavior would be resp["Account"].should.equal("MEMEMEMEME")

This test is being run with pytest. Several other ways to patch env variables have been tried. As seen in the test though the env variable is properly set. That is know bc `os_account_id.should.equal(dummy_account_id)` passes. 

## Reporting Bugs

Please be aware of the following things when filing bug reports:

1. Avoid raising duplicate issues. *Please* use the GitHub issue search feature
   to check whether your bug report or feature request has been mentioned in
   the past.
2. When filing bug reports about exceptions or tracebacks, please include the
   *complete* traceback. Partial tracebacks, or just the exception text, are
   not helpful.
3. Make sure you provide a suitable amount of information to work with. This
   means you should provide:

   - Guidance on **how to reproduce the issue**. Ideally, this should be a
     *small* code sample that can be run immediately by the maintainers.
     Failing that, let us know what you're doing, how often it happens, what
     environment you're using, etc. Be thorough: it prevents us needing to ask
     further questions.
   - Tell us **what you expected to happen**. When we run your example code,
     what are we expecting to happen? What does "success" look like for your
     code?
   - Tell us **what actually happens**. It's not helpful for you to say "it
     doesn't work" or "it fails". Tell us *how* it fails: do you get an
     exception? A hang? How was the actual result different from your expected
     result?
   - Tell us **what version of Moto you're using**, and
     **how you installed it**. Tell us whether you're using standalone server
     mode or the Python mocks. If you are using the Python mocks, include the
     version of boto/boto3/botocore.

   If you do not provide all of these things, it will take us much longer to
   fix your problem.

**Comment by bblommers** [comment]
Hi @bsneider, thanks for raising this - will mark it as a bug.

In case it is helpful, the workaround would be to patch/set the environment variable before Moto is loaded/imported.

**Comment by bsneider** [comment]
> Hi @bsneider, thanks for raising this - will mark it as a bug.
> 
> In case it is helpful, the workaround would be to patch/set the environment variable before Moto is loaded/imported.

Do you have an example @bblommers 

I tried the following and it still failed 
```patch.dict(os.environ, {"MOTO_ACCOUNT_ID": "891514678401"})
from moto import mock_ec2, mock_sns, mock_ssm, mock_sts```

**Comment by bblommers** [comment]
This works for me @bsneider:
```
import boto3
import mock
import os


def test_x():
    with mock.patch.dict(os.environ, {"MOTO_ACCOUNT_ID": "random"}):
        from moto import mock_sts
        with mock_sts():
            client = boto3.client("sts", region_name="us-west-2")
            resp = client.get_caller_identity()
            print(resp)
```
Note that's in a otherwise empty file, with no other imports coming from `__init__.py` for instance.

**Comment by bblommers** [comment]
This feature is now available as of Moto >= 4.0.0.dev3, as part of a broader rework to support multiple accounts - see the documentation here: http://docs.getmoto.org/en/latest/docs/multi_account.html

The full release schedule for Moto 4 can be found here: https://github.com/spulec/moto/issues/5366

Please let us know if you have any feedback.

**Comment by bblommers** [comment]
Moto 4 is now released, so this now works as you would expect. Please raise a new issue if you still encounter any issues.
